docs/Design/Alpha/Data Model.md

```diff
@@ -0,0 +1,215 @@
+# Initial Data Model Design
+This doc details the initial design for the data model. This represents how data will be
+stored persistently for consumption by the API and other backend services. 
+
+## Problem Description
+Straightforwardly, we need a way to represent the data we plan to store such that it is
+easy to read, update, and interpret. The perceived data requirements for this project
+appear to be very simple, so we only need to translate them into a simple database model.
+
+The way that data is represented here does not necessarily reflect how it will look in 
+the API, although they should be similar so that programming with either is consistent. 
+
+## Philosophy
+The data we'll be storing should be easily represented with a relational database model. 
+The model should reflect data owned by "Entities" (**Artist**, **Album**, Song, etc.), and how 
+the entities relates to each other. 
+
+For example, data owned by an **Artist** would include attributes such as the artist's name, 
+and an **Artist** would also "own" multiple **Album**s. This could be represented as a one-to-many
+relationship or a many-to-many relationship if we want to link multiple **Artist**s to a single
+**Album**. Decisions like this ultimately cascade into larger design decisions about the project,
+so it's important to get them right here with future considerations as to not be constantly 
+changing the project's data model over time. Preferable changes to the data model are additions,
+and we should look to avoid drastically modifying existing data as the project grows.
```

Comment:
I agree that this decision has impacts in the long run so personally I think supporting a many-to-many relationship from the get go is preferable. As shown below in the ER diagram intermediary tables easily support a many-to-many relationship. We can control this through the API in case we find that actually supporting many-to-many relationships in the mvp is too much for us initially but we'll always be able add it later if we decide to by using many-to-many intermediary tables.